### PR TITLE
Fix hdf5 history overwrite

### PR DESCRIPTION
**Pull request by PaulJonasJost**
For now, not allowing to write into the history file on already existing id's.

### CODE REVIEW

File: pypesto/objective/history.py [hunk 1]
```diff
@@ -1205,6 +1207,23 @@ def get_time_trace(
...
 
+    def _check_file_id(self, file: str):
+        """
+        Check, whether the id is already existent in the file.
+
+        Parameters
+        ----------
+        file:
+            HDF5 file name.
+        """
```
Comment (dweindl) [inline pypesto/objective/history.py:1218]: Please document return values. It's not obvious for me what "editable" means here.

File: pypesto/objective/history.py [hunk 2]
```diff
@@ -1205,6 +1207,23 @@ def get_time_trace(
...
+        with h5py.File(file, 'a') as f:
```
Comment (dweindl) [inline pypesto/objective/history.py:1219]: Why `'a'`? To check that it's actually writable? If so, potential exceptions should be handled.

File: pypesto/objective/history.py [hunk 3]
```diff
@@ -1205,6 +1207,23 @@ def get_time_trace(
...
+            editable = True
+            if 'history' not in f.keys():
+                return file, editable
+            if self.id in f['history']:
+                editable = False
+            return file, editable
```
Comment (dweindl) [inline pypesto/objective/history.py:1225]:
```suggestion
            return file, ('history' not in f.keys() or self.id not in f['history'])
```

File: pypesto/objective/history.py [hunk 4]
```diff
@@ -936,6 +936,8 @@ def update(
         result: ResultDict,
     ) -> None:
         """See `History` docstring."""
+        if not self.editable:
+            raise ValueError('This id is already used in the history file.')
```
Comment (dweindl) [inline pypesto/objective/history.py:940]:
```suggestion
            raise ValueError(f'ID "{self.id}" is already used in history file "{self.file}".')
```

**Review by jvanhoefer — APPROVED** [review]

**Comment by PaulJonasJost** [comment]
closes #828

**Pull request merged**